core/common/src/test/java/alluxio/collections/LockCacheTest.java

```diff
@@ -82,4 +83,30 @@ public void parallelInsertTest() throws InterruptedException {
     t3.join();
     t4.join();
   }
+
+  @Ignore
+  @Test(timeout = 100000)
+  public void referencedLockTest() throws InterruptedException {
+    try (LockResource resource0 = mCache.get(0, LockMode.READ)) {
```

Comment:
taking the locks doesn't require indentation like this, we could do
```java
lock1 = mCache.get(0, LockMode.READ);
lock2 = mCache.get(50, LockMode.READ);
lock3 = mCache.get(100, LockMode.READ);
for (int i = 0; i < 100; i++) {
  mCache.get(i, LockMode.READ).close();
}

assertSame(lock0.getLock(), mCache.get(0).getLock());
```

We could avoid exposing the locks by giving `LockResource` a `hasSameLock(LockResource other)` method

Reply:
done.